libcylinder/src/signing/secp256k1.rs

```diff
@@ -46,52 +45,38 @@ impl From<ErrorStack> for Error {
     }
 }
 
-pub struct Secp256k1PrivateKey {
-    private: Vec<u8>,
-}
-
-impl Secp256k1PrivateKey {
-    pub fn from_hex(s: &str) -> Result<Self, Error> {
-        hex_str_to_bytes(s)
-            .map(|key_bytes| Secp256k1PrivateKey { private: key_bytes })
-            .map_err(|err| Error::ParseError(err.to_string()))
-    }
-
-    #[cfg(feature = "pem")]
+#[cfg(feature = "pem")]
+impl PrivateKey {
```

Comment:
This should probably go in the key:: module.

Reply:
I'm not sure that these should be on the PrivateKey struct, since they are elliptic curve-specific.  Perhaps these should be a pair of `TryFrom/TryInto` implementations:

```rust
impl TryFrom<EcKey<EcPrivate>> for PrivateKey { ... }
// and
impl TryInto<EcKey<EcPrivate>> for PrivateKey { ... }
```
Or, if we don't want to expose `EcKey`, like we have now, I think we could create `struct PemPrivateKey(String)`, where the contents is the pem key.  We can implement the above `TryFrom/TryInto`pair for this struct instead.

Reply:
I don't agree with the additional structs - PEM is a storage format and irrelevant once we've extracted the key material. Also, we aren't generically supporting PEM (ignoring most of it in fact), just the extraction of key material from within it.

Reply:
If we're not supporting writing the pem format, then I think I'm fine with keeping the `from_pem` method on `PrivateKey`. The implementation does still make assumptions about it being an elliptic curve key, which is not a requirement for the PEM format.